makefiles/features_precheck.inc.mk

```diff
@@ -0,0 +1,15 @@
+# Check if provided features are available
+
+ifneq (,$(filter picolibc,$(FEATURES_PROVIDED)))
+  # Test if picolibc.specs is available
+  ifneq ($(shell $(LINK) -specs=picolibc.specs -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
+    FEATURES_BLACKLIST += picolibc
+ endif
+endif
```

Comment:
```suggestion
# Test if picolibc.specs is available
ifneq ($(shell $(LINK) -specs=picolibc.specs -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
  FEATURES_BLACKLIST += picolibc
endif
```
 Is it as it is now to avoid the shell call? In that case, I would assign it to a variable once and the check that variable, similar to what is done with `OS`.

Reply:
maybe we should but i do not see a huge performance penalty atm.
I will have a look at OS.

Reply:
> So this can be used to opportunistically enable picolibc and fall back to newlib if it's not installed?

Shouldn't the first step be testing it as the default libc in the ci?